### PR TITLE
Resolve issues #12 (gzipped S3) and #13 (readline)

### PR DESCRIPTION
**Pull request by mpenkov**
This pull request includes the changes from https://github.com/piskvorky/smart_open/pull/75 (which I have now closed).

### CODE REVIEW

**Comment by mpenkov** [comment]
Looks like Python 3.x builds are failing. I'm new to Python 3.x, but will see if I can fix these issues during the week.

On a side note, I'm not sure why these problems have come up now as opposed to https://github.com/piskvorky/smart_open/issues/75 - the code is mostly the same. If anyone can shed light onto this, I'd appreciate it greatly.

File: gzipstream/gzipstreamfile.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
+import io
```
Comment (piskvorky) [inline gzipstream/gzipstreamfile.py]: We definitely want to attribute and give credit to the original author / repository / license here!

File: bin/test_s3_gzip.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env python
+import smart_open
+
+URL = "s3://aws-publicdatasets/common-crawl/crawl-data/CC-MAIN-2014-35/\
+segments/1408500800168.29/warc/\
+CC-MAIN-20140820021320-00000-ip-10-180-136-8.ec2.internal.warc.gz"
+
+with smart_open.smart_open(URL) as fin:
```
Comment (piskvorky) [inline bin/test_s3_gzip.py]: Doesn't look like a unit test. If it's some internal test you're using, no need for it to be in repo.

If it's meant to be a unit test, the right place for it is under `tests`, such as inside `tests/test_smart_open.py`.

File: gzipstream/gzipstreamfile.py (hunk 1 above)
Comment (mpenkov) [inline gzipstream/gzipstreamfile.py]: Done.

File: bin/test_s3_gzip.py (hunk 2 above)
Comment (mpenkov) [inline bin/test_s3_gzip.py]: Done.

**Comment by tmylk** [comment]
Looks good. Just waiting for Travis builds to come back after we moved project to another org.

Adding a couple of lines in the CHANGELOG would be good too.

**Comment by mpenkov** [comment]
@tmylk What date and version should I use in the changelog? Are placeholders acceptable?

File: gzipstream/tests/test_gzipstreamfile.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
+from .. import gzipstreamfile
+import unittest
+
+GzipStreamFile = gzipstreamfile.GzipStreamFile
+
+
+class TestSequenceFunctions(unittest.TestCase):
```
Comment (tmylk) [inline gzipstream/tests/test_gzipstreamfile.py]: Just noticed these tests are not run in Travis. What is the best way to make the test runner pick them up?

**Comment by tmylk** [comment]
@mpenkov Thanks for updating the changelog! Once gzipstream tests run, then will merge in.

File: gzipstream/tests/test_gzipstreamfile.py (hunk 3 above)
Comment (mpenkov) [inline gzipstream/tests/test_gzipstreamfile.py]: @tmylk I think moving it into the same subdirectory with the other tests is the easiest way to go. Let me know if this is acceptable for you.

Comment (tmylk) [inline gzipstream/tests/test_gzipstreamfile.py]: That is acceptable. Thanks!

**Comment by tmylk** [comment]
Thanks a lot for this great PR!

**Comment by mpenkov** [comment]
You're welcome. Thank you for such a useful module!

**Pull request merged**